### ISSUE TITLE
No `reveal_type` diagnostic for variables with bare type qualifier annotations

### ISSUE THREAD

**Issue by sharkdp** [microsoft/pyright#9720]
**Describe the bug**

Pyright seems to omit `reveal_type` diagnostics for class variables annotated with a bare `ClassVar` or a bare `Final` (the latter of which makes little sense without an assignment).

This is certainly not a big problem, just a little surprising, as Pyright even shows diagnostics for symbols that are not defined at all (`reveal_type(not_defined_anywhere)`). It's not entirely clear to me if a bare `ClassVar` should even be allowed (it's not mentioned anywhere in the documentation/spec), and a bare `Final` without an assignment raises another diagnostic, but maybe this should be fixed anyway.

**Code or Screenshots**

```python
from typing import ClassVar, reveal_type

class C:
    c: ClassVar

reveal_type(C.c)  # <- no diagnostic here
```

[Playground link](https://pyright-play.net/?pyrightVersion=1.1.392&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCANgIYDOZAaiSADRQgCmAboyUQPrwKMBQvAY1IVCALl5RJUAaMLCqNfk1bsuiRgAoCAOgEBKSQGIoAHgC0UFGCgATJCTRWyMJAKgALRk15A)

**Version information**

Pyright 1.1.392

**Comment by erictraut** [comment]
Thanks for the bug report.

Pyright is not correctly handling a "bare" `ClassVar` qualifier. It's not clear from the spec whether this should be an error condition or treated implicitly as `ClassVar[Any]`. It appears that mypy does the latter, but I could make an argument for both behaviors. It would be good for the typing spec to be explicit about this. In any case, pyright should handle this better.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.393.
